packages/compass-editor/src/codemirror/query-autocompleter.test.ts

```diff
@@ -0,0 +1,87 @@
+import { EditorView } from '@codemirror/view';
+import { EditorState, EditorSelection } from '@codemirror/state';
+import type {
+  CompletionSource,
+  CompletionResult,
+} from '@codemirror/autocomplete';
+import { CompletionContext } from '@codemirror/autocomplete';
+import { expect } from 'chai';
+// import { languages } from '../json-editor';
+import { createQueryAutocompleter } from './query-autocompleter';
+
+const setupCompleter = <T extends (...args: any[]) => CompletionSource>(
+  completer: T
+) => {
+  const el = window.document.createElement('div');
+  window.document.body.appendChild(el);
+  const editor = new EditorView({
+    state: EditorState.create({
+      doc: '',
+    }),
+    selection: EditorSelection.single(0),
+    // TODO: doesn't look like extensions work when inside jsdom. need to sort
+    // it out to be able to better test completers
```

Comment:
We meaningfully use tokens for autocompletion only in aggregation autocompleter, so this probably can wait for now